### PR TITLE
community[minor]: feat: datastore message store

### PR DESCRIPTION
**Pull request by saul-jb**
This PR adds a chat message history wrapper for stores implementing the [datastore interface](https://github.com/ipfs/js-stores/tree/main/packages/interface-datastore).

This datastore interface is used to make modularized datastores used in the IPFS ecosystem. This wrapper means we can leverage the existing (and future) implementations of datastores as message storage.

Here are some of the datastores implementations that can be leveraged by this:

- [datastore-core](https://github.com/ipfs/js-stores/blob/main/packages/datastore-core) Datastore in-memory implementation.
- [datastore-fs](https://github.com/ipfs/js-stores/blob/main/packages/datastore-fs) Datastore implementation with file system backend.
- [datastore-idb](https://github.com/ipfs/js-stores/blob/main/packages/datastore-idb) Datastore implementation with IndexedDB backend.
- [datastore-level](https://github.com/ipfs/js-stores/blob/main/packages/datastore-level) Datastore implementation with level(up|down) backend
- [datastore-s3](https://github.com/ipfs/js-stores/blob/main/packages/datastore-s3) Datastore implementation backed by s3.

Example usage:
```javascript
import { FsDatastore } from 'datastore-fs'
import { DatastoreChatMessageHistory } from '@langchain/community/stores/message/datastore'

const datastore = new FsDatastore('path/to/store')
const sessionId = 'my-session'

const history = new DatastoreChatMessageHistory({ datastore, sessionId })
```

### CODE REVIEW

File: libs/langchain-community/src/stores/message/datastore.ts
Comment (jacoblee93) [inline libs/langchain-community/src/stores/message/datastore.ts:1]: I know this is the name, but can we call this `ipfs_datastore` instead to be less generic?

File: libs/langchain-community/package.json [hunk 1]
```diff
@@ -219,6 +220,7 @@
     "@zilliz/milvus2-sdk-node": ">=2.2.7",
     "better-sqlite3": "^9.4.0",
     "cassandra-driver": "^4.7.2",
+    "cborg": "^4.1.1",
```
Comment (jacoblee93) [inline libs/langchain-community/package.json:223]: Could you add all these deps as peer deps and mark them as optional?

https://github.com/langchain-ai/langchainjs/blob/main/.github/contributing/INTEGRATIONS.md#third-party-dependencies

**Comment by jacoblee93** [comment]
Thanks! Can you also add docs?

File: libs/langchain-community/src/stores/message/datastore.ts
Comment (saul-jb) [reply libs/langchain-community/src/stores/message/datastore.ts:1]: Done in 661c81a571a94e470afdf0e7445c166ab6b6a3fd

File: libs/langchain-community/package.json (hunk 1 above)
Comment (saul-jb) [reply libs/langchain-community/package.json:223]: Fixed: 21a55f1efb45f92f813614c3963c51a97f13138f

**Comment by saul-jb** [comment]
> Thanks! Can you also add docs?

Added docs in 20c52eea7fe92a38d8eeb4981a4d94b36d321f80.

File: docs/core_docs/docs/integrations/chat_memory/ipfs_datastore.mdx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+- [datastore-idb](https://github.com/ipfs/js-stores/blob/main/packages/datastore-idb) Datastore implementation with IndexedDB backend.
+- [datastore-level](https://github.com/ipfs/js-stores/blob/main/packages/datastore-level) Datastore implementation with level(up|down) backend
+- [datastore-s3](https://github.com/ipfs/js-stores/blob/main/packages/datastore-s3) Datastore implementation backed by s3.
+
+## Usage
+
+```typescript
+// Replace FsDatastore with the IPFS Datastore of your choice.
+import { FsDatastore } from 'datastore-fs'
+import { DatastoreChatMessageHistory } from '@langchain/community/stores/message/datastore'
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/chat_memory/ipfs_datastore.mdx:37]: Update examples here - you could also move this to `examples/` and import to get linting active

**Comment by jacoblee93** [comment]
Thank you! Small nit that I can get to later today.

**Comment by jacoblee93** [comment]
Patched and should be good to go, thanks!

**Pull request merged**